### PR TITLE
Fix/325 consistent price format

### LINKED ISSUE #325: Ensure consistent price format for different product types.
### Describe the bug:

Right now some of the prices are added to the feed in non-decimal format like 10USD ( simple product ) and 10.00USD ( variable product ). Accodording to the documentation it should always be the second format:

image

We probably just need to ensure to run `wc_format_decimal( $price, wc_get_price_decimals() );` over all of the functions.

#### Expected behavior
All prices are in format according to the documentation.

# Todo

- [ ] Update the code to conform to the specification
- [ ] Update UT ( they should fail after the code change )

### PR DESCRIPTION
**Pull request by alexiglesias31**
### Changes proposed in this Pull Request:

Closes #325 .

Add the prices to the feed with the proper decimal format like 10.00USD (second format of the screenshot).
UT was updated to success with the new changes.

### Screenshots:

149070840-9fa2244d-f306-4b30-8626-73ecca462068

### Detailed test instructions:

1. Create a product with a non-decimal price (like 10USD).
2. Verify that the price in the feed is in decimal format (like 10.00USD).
3. Verify that UT are success with the new changes.

### Changelog entry

> Update - Ensure that the prices are formatted in a consistent way across the feed file.

### CODE REVIEW

**Comment by budzanowski** [comment]
In WooCommerce the number of decimal points is configurable in the settings. Since the Feed has specific requirements the code may be forced to be indifferent to the system settings. The number of decimal points settings is available at WooCommerce > Settings > General ( bottom of the tab )
image 

We should confirm with Pinterest if a variable precision is acceptable. I am mentioning this because we have seen sites that are using for example USD ( so they should use 2 decimal points according to ISO-4271 ) but site owners want to use 0 numbers for precision - for example, because they sell higher-priced products and they want to just show 1000 USD as a price sitewide. If we would be forced to use 2 number precision in such cases then this, in turn, would mean that we can't really rely on the `wc_get_price_decimals` function.

**Review by ksere — CHANGES_REQUESTED** [review]
Great feedback, thanks Bartosz. 

I 'd say since the document is clear enough we go with the default decimal points for the given currency. We should be able to get that from [locale-info.php](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/i18n/locale-info.php).

File: src/ProductsXmlFeed.php [hunk 1]
```diff
@@ -414,7 +414,16 @@ private static function get_taxonomies( $product_id ) {
 	}
 
 	/**
-	 * Fetch shipping object.
+	 * Get locale currency decimals
+	 */
+	private static function get_currency_decimals() {
```
Comment (ksere) [inline src/ProductsXmlFeed.php:419]: This won't produce the desired output if for example my base country is US and my currency is JPY.
We need to get the decimals from that array based on the currency returned by `get_woocommerce_currency()`

**Review by ksere — CHANGES_REQUESTED** [review]

File: src/ProductsXmlFeed.php [hunk 2]
```diff
@@ -414,7 +414,22 @@ private static function get_taxonomies( $product_id ) {
...
+		$locale_info = include WC()->plugin_path() . '/i18n/locale-info.php';
```
Comment (ksere) [inline src/ProductsXmlFeed.php]: Maybe move the `include` in the conditional as well so that it doesn't load up the array when not needed.

**Review by ksere — APPROVED** [review]

**Comment by budzanowski** [comment]
This looks great, approving. Could you please add one additional test where you use different currency - to highlight that the formatting for different decimal points works?

**Review by budzanowski — APPROVED** [review]

**Comment by alexiglesias31** [comment]
Done @budzanowski. Added the unit test with JPY (that has 0 decimals as default).

**Comment by budzanowski** [comment]
I have updated the changelog entry in the PR description. Please make sure to have it as it is used by our release tool to automatically populate the changelog.

**Pull request merged**